### PR TITLE
Pip editable install fix

### PR DESCRIPTION
**Pull request by lizzcondrey**
# Description

Clearing break for doing an editable pip install within the pipeline as part of the pyproject.toml migraion

# Linked Items:

N/A

### Fixed

- Fixed site deployment with editable package install. 
- Standardized pip install step between workflows

# Checklist:

- [ ] I updated project documentation to reflect my changes.
- [ ] My changes generate no new warnings.
- [ ] I updated new and existing unit tests to account for my changes.
- [ ] I linked the associated item(s) to be closed.
- [ ] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

File: workflow-templates/python-cross-platform-support.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
           python-version: 3.11
 
       - name: Install ${{ github.event.repository.name }} Package
-        run: python -m pip install -e .[all]
+        run: pip install -e .[all]
```
Comment (bunchw) [inline workflow-templates/python-cross-platform-support.yml:27]: Why is this changed?  And should it also be changed in the core aac repo?

Comment (lizzcondrey) [reply workflow-templates/python-cross-platform-support.yml:27]: From testing I did in my forked repo, it wasn't necessary and this is the only spot that our pip install had the preface of "python -m". So, removed it to be consistent, and not worried about updating it in core with this effort because we have a task for pulling all of the new template workflows here shortly. But till we are there, we need main to be working correctly so just applied the doc deployment change.

**Review by bunchw — APPROVED** [review]
